### DISCUSSION TITLE
why can I not access std::stop_token?

### DISCUSSION THREAD

**Discussion by patrick99e99** [emscripten-core/emscripten#22896] [General]
I am using emcc with `-std=c++23` and am getting:

```
emulator.cpp:13:8: error: no type named 'stop_token' in namespace 'std'
   13 |   std::stop_token token;
      |   ~~~~~^
```

**Comment by sbc100** [comment]
It looks like `stop_token` in marked as experimental in the LLVM 18 version of libc++.  See https://github.com/emscripten-core/emscripten/blob/2e84cfdc8b7a5e04fc3d2f8a0e09ac98e2a88a0f/system/lib/libcxx/include/__config#L507-L514.

Looks like you can enable it at your own risk by defining `_LIBCPP_ENABLE_EXPERIMENTAL`.

**Comment by sbc100** [comment]
It looks like this is also still experimental in the LLVM 19 version of libc++ so even once we fix #22909 this will still be an issue.
